first-party/jib-native-image-extension-gradle/src/test/java/com/google/cloud/tools/jib/gradle/extension/nativeimage/JibNativeImageExtensionTest.java

```diff
@@ -0,0 +1,237 @@
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.ContainerBuildPlan;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.api.buildplan.FileEntry;
+import com.google.cloud.tools.jib.api.buildplan.FilePermissions;
+import com.google.cloud.tools.jib.gradle.ContainerParameters;
+import com.google.cloud.tools.jib.gradle.JibExtension;
+import com.google.cloud.tools.jib.gradle.extension.GradleData;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger;
+import com.google.cloud.tools.jib.plugins.extension.JibPluginExtensionException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.gradle.api.Project;
+import org.gradle.internal.extensibility.DefaultConvention;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link JibNativeImageExtension}. */
+@RunWith(MockitoJUnitRunner.class)
+public class JibNativeImageExtensionTest {
+
+  @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Mock private ExtensionLogger logger;
+
+  @Mock private DefaultConvention defaultConvention;
+
+  @Mock private Project project;
+  private final GradleData gradleData = () -> project;
+  @Mock private JibExtension jibPlugin;
+  @Mock private ContainerParameters jibContainer;
+
+  private static FileEntriesLayer buildLayer(String layerName, Path... paths) {
+    FileEntriesLayer.Builder builder = FileEntriesLayer.builder().setName(layerName);
+    for (Path path : paths) {
+      builder.addEntry(path, AbsoluteUnixPath.get("/dest/" + path.getFileName()));
+    }
+    return builder.build();
+  }
+
+  private static List<String> layerToExtractionPaths(FileEntriesLayer layer) {
+    return layer.getEntries().stream()
+        .map(layerEntry -> layerEntry.getExtractionPath().toString())
+        .collect(Collectors.toList());
+  }
+
+  @Before
+  public void setUp() {
+    when(project.getExtensions()).thenReturn(defaultConvention);
+    when(project.getExtensions().findByType(JibExtension.class)).thenReturn(jibPlugin);
+    when(jibPlugin.getContainer()).thenReturn(jibContainer);
+
+    when(project.getBuildDir()).thenReturn(tempFolder.getRoot());
+  }
+
+  @Test
+  public void testGetExecutableName_property() {
+    Map<String, String> properties = Collections.singletonMap("imageName", "theExecutable");
+    assertEquals(
+        Optional.of("theExecutable"),
+        JibNativeImageExtension.getExecutableName(jibContainer, properties));
```

Comment:
```suggestion
    assertThat(
        JibNativeImageExtension.getExecutableName(jibContainer, properties)).isEqualTo(Optional.of("theExecutable"));
```
We will have to add the following to the import statements: `import static com.google.common.truth.Truth.assertThat`

Reply:
Same below

Reply:
Are you wanting to get rid of all the assertEquals() in favour of assertThat()?